pkg/scheduler/factory_test.go

```diff
@@ -284,124 +304,146 @@ func TestDefaultErrorFunc(t *testing.T) {
 			SecurityContext:               &v1.PodSecurityContext{},
 		},
 	}
+	testPodUpdated := testPod.DeepCopy()
+	testPodUpdated.Labels = map[string]string{"foo": ""}
 
 	nodeBar, nodeFoo :=
 		&v1.Node{ObjectMeta: metav1.ObjectMeta{Name: "bar"}},
 		&v1.Node{ObjectMeta: metav1.ObjectMeta{Name: "foo"}}
 
-	testPodInfo := &framework.QueuedPodInfo{Pod: testPod}
 	client := fake.NewSimpleClientset(&v1.PodList{Items: []v1.Pod{*testPod}}, &v1.NodeList{Items: []v1.Node{*nodeBar}})
+	// podIndexer := cache.NewIndexer(cache.MetaNamespaceKeyFunc, cache.Indexers{cache.NamespaceIndex: cache.MetaNamespaceIndexFunc})
+	podLister := corelisters.NewPodLister(newIndexer(func(namespace, name string) (interface{}, error) {
```

Comment:
can't we create an informer from the fake client and then get the lister from that informer instead of doing all this?

Reply:
yes, that's possible. In that case, we should manually operate the podInformer.Store.